### PR TITLE
Cswinrt and windows50

### PR DESCRIPTION
**Pull request by wli3**
Now that we have all the changes flowed in

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs [hunk 1]
```diff
@@ -154,6 +154,141 @@ public void It_does_not_download_desktop_runtime_packs_on_unix()
...
+        {
+            var testProject = new TestProject()
+            {
+                Name = "WinExeOutput",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true,
+                IsExe = true,
+            };
+            testProject.AdditionalProperties[propName] = propValue;
+            testProject.AdditionalProperties["TargetPlatformVersion"] = "7.0"; // Ensure TargetPlatformVersion is set so we can build with a TargetPlatformIdentifier
```
[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:172]: Do we?

File: src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs [hunk 2]
```diff
@@ -154,6 +154,141 @@ public void It_does_not_download_desktop_runtime_packs_on_unix()
...
+                .Should()
+                .Pass();
+
+            var values = getValuesCommand.GetValues();
+            values.Count.Should().Be(1);
+            values.First().Should().Be(expectedOutputType);
+        }
+
+        [WindowsOnlyFact]
+        public void It_builds_on_windows_with_the_windows_desktop_sdk_5_0_with_ProjectSdk_set()
```
[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:188]: "Microsoft.NET.Sdk.WindowsDesktop" still works

File: src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs [hunk 3]
```diff
@@ -154,6 +154,141 @@ public void It_does_not_download_desktop_runtime_packs_on_unix()
                 .NotHaveSubDirectories($"runtime.{Rid}.microsoft.windowsdesktop.app");
         }
 
+        [Theory]
+        [InlineData("net5.0", "TargetPlatformIdentifier", "Windows", "WinExe")]
```
[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:158]: That's just a bad diff from github. I didn't change this test. I think we should make a separate issue for it

[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:158]: @sfoslund do you mind to follow up on that?

File: src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs]: Removed

File: src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:158]: create a separated issue https://github.com/dotnet/sdk/issues/12440

[the remark this answers is not in the document]
Comment (sfoslund) [reply src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs:158]: Sure, PR is out here: #12448

**Review by dsplaisted — APPROVED** [review]

**Pull request merged**